### PR TITLE
RATSTAT DAG 

### PR DESCRIPTION
**Pull request by hunterowens**
currently, Rat trap statistic information is kept in a group on ArcGIS via the ESRI collector app. 

This DAG pulls the data nightly and replaces it in the `public-health` schema of the data warehouse.

### CODE REVIEW

**Comment by hunterowens** [comment]
cc @brendanbailey - can you take a look at this script / test it?

File: dags/public-health/ratstat.py [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+from datetime import datetime, timedelta
+import logging
+pg_conn = BaseHook.get_connection('postgres_default') 
+
+def ratstat_loader():
+    """
+    load the ratstat data 
+    into a postgres DB
+    """ 
+    hub = arcgis.gis.GIS(url='https://lahub.maps.arcgis.com',
```
Comment (ian-r-rose) [inline dags/public-health/ratstat.py]: Use airflow credentials db here.

File: dags/public-health/ratstat.py [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+                         username=os.environ.get('LAHUB_USERNAME'),
+                         password=os.environ.get('LAHUB_PASSWORD'))
+    rat_stat_group = arcgis.gis.Group(hub, 
+                                      '7f3d66478dd846598e76a8e334a03988') #this is the groupid of the ratstat group
```
Comment (ian-r-rose) [inline dags/public-health/ratstat.py]: Can we add this hash as a const variable? Makes it more legible.

File: dags/public-health/ratstat.py [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+                         username=os.environ.get('LAHUB_USERNAME'),
+                         password=os.environ.get('LAHUB_PASSWORD'))
+    rat_stat_group = arcgis.gis.Group(hub, 
+                                      '7f3d66478dd846598e76a8e334a03988') #this is the groupid of the ratstat group       
+    logging.info("Logged into hub and accessed rat stat group.")
+    content = rat_stat_group.content()
+    feature_layers = []
+    for item in content:
+        try:
+            [feature_layers.append(feature_layer) for feature_layer in item.layers]
```
Comment (ian-r-rose) [inline dags/public-health/ratstat.py]: Can we make this a for loop instead of a list comprehension?

File: dags/public-health/ratstat.py [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+    logging.info("Connecting to DB")
+    user = pg_conn.login
+    password = pg_conn.get_password()
+    host = pg_conn.host
+    dbname = pg_conn.schema
+    logging.info(f"Logging into postgres://-----:----@{host}:5432/{dbname}")
+    conn = create_engine(f'postgres://{user}:{password}@{host}:5432/{dbname}'')
+    for k,table in dfs.items(): 
+        table['SHAPE'] = str(table['SHAPE'])
+        table.to_sql(k,
```
Comment (ian-r-rose) [inline dags/public-health/ratstat.py]: We can make this into a PostGIS Geometry type using GeoAlchemy

**Comment by brendanbailey** [comment]
Also .layers will only get the layers within the webmap. There are a couple of tables associated with the layers, so you will need to use .tables as well

File: dags/public-health/ratstat.py (hunk 4 above)
Comment (hunterowens) [reply dags/public-health/ratstat.py]: Do you have a code segment to point me to here?

Comment (ian-r-rose) [reply dags/public-health/ratstat.py]: Yeah, [this](https://github.com/CityOfLosAngeles/aqueduct/blob/2dcb60b34a6109580f115a2b22ee1c117d8547ad/dags/public-health/care-311-to-postgres.py#L47-L62) dag does it.

**Comment by ian-r-rose** [comment]
Closing as stale

**Pull request closed without merging**